pkg/kubelet/status/manager.go

```diff
@@ -236,7 +241,9 @@ func (m *manager) syncBatch() error {
 		statusPod, err = m.kubeClient.Pods(pod.Namespace).UpdateStatus(statusPod)
 		if err == nil {
 			glog.V(3).Infof("Status for pod %q updated successfully", kubeletUtil.FormatPodName(pod))
-
+			if psInfo, ok := m.podStatusInfos[pod.UID]; ok {
```

Comment:
You didn't acquire the lock and should not be modifying the map directly here. How about adding another method to flip the `flushed` bit?